rtcpeerconnection.js

```diff
@@ -636,11 +640,15 @@ module.exports = function(window, edgeVersion) {
           delete p.rtx;
         });
       }
-      params.encodings = transceiver.recvEncodingParameters;
+      if (transceiver.recvEncodingParameters.length) {
+        params.encodings = transceiver.recvEncodingParameters;
+      }
       params.rtcp = {
-        cname: transceiver.rtcpParameters.cname,
         compound: transceiver.rtcpParameters.compound
       };
+      if (transceiver.rtcpParameters.cname) {
```

Comment:
for this to happen the SDP in setRemoteDescription would not contain an a=ssrc: cname line. What SDP are you trying to use this with?

Reply:
Regarding the missing 'a=ssrc: cname ' I did some reading, and agree that it is a "MUST" have based on reading the RFC when using SDP with WebRTC. Sadly, Asterisk seems to have only added this SDP feature in Asterisk 15, and many people are still using Asterisk 11 and 13 - I am just now making the transition myself.

Only EDGE seems to actually care - Firefox and Chrome are perfectly happy, but they are using the older interface.

Reply:
I think omitting it as you do here is correct -- see http://draft.ortc.org/#rtcrtcpparameters*
Not quite clear what happens then but if you get audio... that should be ok.